### PR TITLE
Drop use of maap-py for obtaining S3 credentials

### LINKED ISSUE #14: Make S3 credentials caching/refresh logic more robust
Our logic for caching and refreshing S3 credentials does not make use of similar logic already built into the boto3 API. Thus, our logic is both redundant, and perhaps less robust.

We want to take advantage of the built-in logic to eliminate this redundant (and perhaps more brittle) logic.

For reference, see the following:

- https://stackoverflow.com/questions/61899028/where-can-i-find-the-documentation-for-writing-custom-aws-credential-provider-us (code example as a good starting point, but perhaps could be improved a bit)
- https://github.com/boto/botocore/blob/develop/botocore/credentials.py

### PR DESCRIPTION
**Pull request by chuckwondo**
S3 credentials are now obtained via a role using EC2 instance metadata. In addition, removed HTTP retry logic since all granules are now obtained via S3 only.

Fixes #14

### CODE REVIEW

**Comment by wildintellect** [comment]
I'm generally in favor of this for actual MAAP usage, but have concerns about removing the functionality to operate in a different way given the possibility of wanting to us this outside of MAAP, e.g. VEDA.
That would favor dropping maap-py but I wonder if we can more generically make AWS credentials modular to allow several options for obtaining (e.g. EarthAccess)

**Comment by chuckwondo** [comment]
> I'm generally in favor of this for actual MAAP usage, but have concerns about removing the functionality to operate in a different way given the possibility of wanting to us this outside of MAAP, e.g. VEDA. That would favor dropping maap-py but I wonder if we can more generically make AWS credentials modular to allow several options for obtaining (e.g. EarthAccess)

Yes, there is a separate issue for refactoring for use outside of MAAP. When I work on that issue, I'll sort out how to make this modular. At the moment, I feel this simplification will better enable introducing modularization later. There will be less cruft to deal with.

**Comment by chuckwondo** [comment]
@wildintellect and @jjfrench, I've taken this PR out of Draft. It is now ready for review. I registered `gedi-subset:issue14-auto-refresh-s3-creds` and confirmed that it works.

File: bin/build-dps [hunk 1]
```diff
@@ -17,4 +17,4 @@ set -euo pipefail
 
 basedir=$(dirname "$(readlink -f "$0")")
 
-"${basedir}/build" --no-dev
+AWS_PROFILE=maap-data-reader "${basedir}/build" --no-dev
```
Comment (jjfrench) [inline bin/build-dps:20]: We need to run the build with the AWS_PROFILE as well?

Comment (chuckwondo) [reply bin/build-dps:20]: Actually, upon further experimentation, I don't think we need to specify the profile in either place. Let me remove both appearances and reregister and rerun to confirm.

Comment (chuckwondo) [reply bin/build-dps]: @jjfrench, everything worked without the explicit profile.

**Review by jjfrench — APPROVED** [review]
Lookin good, thanks!

**Pull request merged**